lib/adf_diag.py

```diff
@@ -510,12 +513,24 @@ def call_ncrcat(cmd):
 
             #Loop over CAM history variables:
             list_of_commands = []
-            for var in self.diag_var_list:
+            vars_to_derive = []
+            #create copy of var list that can be modified for derivable variables
+            diag_var_list = self.diag_var_list
+            for var in diag_var_list:
                 if var not in hist_file_var_list:
-                    msg = f"WARNING: {var} is not in the file {hist_files[0]}."
-                    msg += " No time series will be generated."
-                    print(msg)
-                    continue
+                    vres = res[var]
```

Comment:
I think you'll want to check here if `var` is actually in the `variable_defaults` dictionary, and if not then return an empty dictionary.  So something like this:

```suggestion
                    vres = res.get(var, {})
```

Reply:
Done.